pkg/drivers/vfkit/vfkit.go

```diff
@@ -136,42 +136,24 @@ func (d *Driver) GetIP() (string, error) {
 	return d.IPAddress, nil
 }
 
-func checkPid(pid int) error {
-	process, err := os.FindProcess(pid)
-	if err != nil {
-		return err
-	}
-	return process.Signal(syscall.Signal(0))
-}
-
 func (d *Driver) GetState() (state.State, error) {
-	if _, err := os.Stat(d.pidfilePath()); err != nil {
-		return state.Stopped, nil
-	}
-	p, err := os.ReadFile(d.pidfilePath())
+	pidfile := d.pidfilePath()
+	pid, err := process.ReadPidfile(pidfile)
 	if err != nil {
-		return state.Error, err
-	}
-	pid, err := strconv.Atoi(strings.TrimSpace(string(p)))
-	if err != nil {
-		return state.Error, err
-	}
-	if err := checkPid(pid); err != nil {
-		// No pid, remove pidfile
-		os.Remove(d.pidfilePath())
+		if !errors.Is(err, os.ErrNotExist) {
+			return state.Error, err
+		}
 		return state.Stopped, nil
 	}
-	ret, err := d.GetVFKitState()
+	exists, err := process.Exists(pid, "vfkit")
 	if err != nil {
 		return state.Error, err
 	}
-	switch ret {
-	case "running", "VirtualMachineStateRunning":
-		return state.Running, nil
-	case "stopped", "VirtualMachineStateStopped":
+	if !exists {
+		os.Remove(pidfile)
```

Comment:
doesm't this return error ? if not wanna return error at least we could log it for the verbose log

Reply:
If we treat this as an error, stop() or kill() may never complete, even when we know that the process is not running. So I think we should not fail the call, but we should log a debug message.

Reply:
Resolved by logging the error. We have 3 instances of this, one existing (GetState()), and 2 new (Stop(), Kill()). Current version log the error in all instances.